app/src/ui/history/merge-call-to-action-with-conflicts.tsx

```diff
@@ -95,37 +237,55 @@ export class MergeCallToActionWithConflicts extends React.Component<
     return null
   }
 
-  private renderLoadingMergeMessage() {
+  private renderLoadingMessage() {
     return (
       <div className="merge-message merge-message-loading">
-        Checking for ability to merge automatically...
+        Checking for ability to {this.state.selectedOperation.toLowerCase()}{' '}
+        automatically...
       </div>
     )
   }
 
-  private renderCleanMergeMessage(
-    currentBranch: Branch,
-    branch: Branch,
-    count: number
-  ) {
-    if (count > 0) {
-      const pluralized = count === 1 ? 'commit' : 'commits'
+  private renderCleanMessage(currentBranch: Branch, branch: Branch) {
+    if (this.commitCount <= 0) {
+      return null
+    }
+
+    const pluralized = this.commitCount === 1 ? 'commit' : 'commits'
+
+    if (this.state.selectedOperation === MultiCommitOperationKind.Rebase) {
       return (
         <div className="merge-message">
-          This will merge
-          <strong>{` ${count} ${pluralized}`}</strong>
-          {` from `}
+          This will update <strong>{currentBranch.name}</strong>
+          {` by applying its `}
+          <strong>{` ${this.commitCount} ${pluralized}`}</strong>
```

Comment:
This space shouldn't be needed, right?
```suggestion
          <strong>{`${this.commitCount} ${pluralized}`}</strong>
```

Reply:
Just double checked and it's rendered correctly anyway, I don't understand HTML but ok 🤣